lms/djangoapps/grades/tasks.py

```diff
@@ -130,6 +131,14 @@ def _has_database_updated_with_new_score(
     if score is None:
         # score should be None only if it was deleted.
         # Otherwise, it hasn't yet been saved.
+        if not score_deleted:
```

Comment:
Will need to rebase on my PR that changes this code: https://github.com/edx/edx-platform/pull/14360